presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java

```diff
@@ -483,6 +488,13 @@ private void splitFinished(PrioritizedSplitRunner split)
 
     private synchronized void scheduleTaskIfNecessary(TaskHandle taskHandle)
     {
+        //Worker skip processing split if jvm heap usage crosses configured threhold
+        //Helps reduce memory pressure on the worker and avoid OOMs
+        if (isLowMemory()) {
+            log.debug("Skip task scheduling due to low memory");
```

Comment:
Would it be better to add this logging in the isLowMemory method so that relevant extra stats can be logged as well there, providing more insight into that decision. 

Also maybe the counter update there as well to DRY out this code

Reply:
Both places where we are checking low memory, put a different logging to help understand which flow is being skipped. This will help in debugging issues in the future and that's why it's a `debug` log and not `info` log.